synthtool/languages/java.py

```diff
@@ -109,6 +111,46 @@ def fix_grpc_headers(grpc_root: Path, package_name: str) -> None:
     )
 
 
+def latest_maven_version(group_id: str, artifact_id: str) -> Optional[str]:
+    """Helper function to find the latest released version of a Maven artifact.
+
+    Fetches metadata from Maven Central and parses out the latest released
+    version.
+
+    Args:
+        group_id (str): The groupId of the Maven artifact
+        artifact_id (str): The artifactId of the Maven artifact
+
+    Returns:
+        The latest version of the artifact as a string or None
+    """
+    group_path = "/".join(group_id.split("."))
+    url = (
+        f"https://repo1.maven.org/maven2/{group_path}/{artifact_id}/maven-metadata.xml"
+    )
+    response = requests.get(url)
+    response.raise_for_status()
+    return version_from_maven_metadata(response.content.decode("utf-8"))
```

Comment:
Are you sure it will be utf-8?  Can we check the content-encoding response header?

Reply:
Done. Using `response.text` which uses requests' best guess at the content encoding by parsing the response headers.